packages/overlay/src/loader.ts

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { TriggerInteractions, OverlayOptions } from './overlay-types';
+
+export const openOverlay = async (
+    target: HTMLElement,
+    interaction: TriggerInteractions,
+    content: HTMLElement,
+    options: OverlayOptions
+): Promise<() => void> => {
+    const { Overlay } = await import('./overlay.js');
+    return await Overlay.open(target, interaction, content, options);
+};
```

Comment:
This centralized the load pattern for lazy `openOverlay` functionality into an easily to leverage export. This will make documenting this `Overlay` API much easier, a task I've avoided a little until now. 😬 

As per @adixon-adobe:
> The default behavior adds network latency to load the overlay script the first time you open a dropdown. On high latency connections this won't be a nice experience.

This is a valid point. However, there is no avoiding latency, it's only a question of where you put it. We can block initial load, but making this eager loading, and in high latency environments this would extend that blocking. Having it lazy allows for developers to really choose how to tune this in a way that can't be broken apart after the fact. In this way tools like `rel="preload"`, `rel="modulepreload"`, service workers etc. can be leveraged to ensure that the code is available (and even parsed if you want it to be) before it's requested without effecting the initial load of the page/functionality. The service worker can then also be used to prevent that load/latency hit in the future. I'd like to prefer having high control in this area, but I can be convinced (especially for the 10KB we're talking about) that it is overkill.

Reply:
I think having high control here is good overall, but I also don't like the default behavior.

It's easy for me to reason about importing `sp-dropdown` and how to structure our lazy-loading so that we only load that code when it's needed. The current default behavior here requires me to know that dropdown uses overlay and that overlay has an async import to have control over it. And given no control, there's a hidden network error condition in the implementation (if that async load fails, the dropdown won't open, and it's likely I wouldn't have any error handling for that scenario).

While I see the appeal of the current approach (you don't need to add a separate import for the async load case), I also like the idea of having async loading like this be "opt-in".

Reply:
The hope is that you don't have to know anything and that with a quality webpack or Rollup set up things work the best way they possibly could. 🤞 😬 

Have you seen a pattern when you're able to opt-in to lazy loading? Seems like what you're asking for, but I can't think of a way to do it. The minute you move to `import { Overlay } from '../overlay'` you trapped with a static dependency and the builder magic to make that go away is intense. A implementor _could_ move that up on level and lazily load dropdown for themselves, but that still means they're loading code a user might not use.

The only sort of practical approach that I can see if having a dropdown that is eager and another that is lazy, but then you're either trading in multiple tag names or the high likely hood of double (read one that throws an error or one that's difficult to ensure is the one you are expecting) registrations.

Reply:
Well I always try to think worst-case for anything async which is why I bring up the error condition 😁 

Just to check my understanding though, say you have 2 imports that register `sp-dropdown`. `@spectrum-web-components/dropdown/sp-dropdown` and `@spectrum-web-components/dropdown/sp-dropdown-async`. And effectively the contents of the former is:

```
import '@spectrum-web-components/overlay/active-overlay';
import './sp-dropdown-async';
```

Would there be any issue with that technically? (there's an obvious maintenance issue with this approach, particularly if it becomes more widespread than overlay)

Reply:
@adixon-adobe this version uses `@spectrum-web-components/dropdown/sp-dropdown` and `@spectrum-web-components/dropdown/sync/sp-dropdown` as I feel pretty strongly that the default should be the version that's most "optimal" on the first load. What's more, while build tooling can take away async with various settings, nothing can add it when it's not, so having it as the default seems like the best way for users to fully understand the power of these elements. If anyone else has feelings on the default of async vs sync, let's chat, happy to hear use cases that this might endanger.

This update effects the `@spectrum-web-components/overlay/overlay-trigger` registration as well, adding `@spectrum-web-components/overlay/sync/. Both packages now have extended READMEs to reference this new feature, and the exports/side-effects in package.json have been updated as well.

Reply:
This seems reasonable (and with some tweaks to our linter, something we can enforce either way) -- thanks!

I do disagree about the default behavior here (or at least am not fully convinced), but definitely not as strongly as you. For me it's about making the default the simplest approach to understand, with the most expected behavior. I think clients that aren't zealous about keeping their package size to an absolute minimum might be surprised by this.

The fact that there's a side-effect of registering a shadow DOM on the `body` tag from that async import also makes this case feel a little less good to me.

I'm definitely not gonna hold things up here, and will be curious to hear what other projects think of this approach. We have the knowledge/tooling to do what we want here, and overall I *really* appreciate the focus on minimal footprint.